src/tools/user-agent-parser/user-agent-parser.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <div>
+    <n-form-item label="User agent string">
+      <n-input
+        v-model:value="ua"
+        type="textarea"
+        placeholder="Put your user-agent here..."
+        clearable
+        :autosize="{ minRows: 2 }"
+      />
+    </n-form-item>
+
+    <user-agent-result-cards :user-agent-info="userAgentInfo" :sections="sections" />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, PropType, ref } from 'vue';
+import { UAParser } from 'ua-parser-js';
+import { withDefaultOnError } from '@/utils/defaults';
+import { Adjustments, Browser, Cpu, Devices, Engine } from '@vicons/tabler';
+import UserAgentResultCards from './user-agent-result-cards.vue';
+
+const ua = ref(navigator.userAgent as string);
+
+// If not input in the ua field is present return an empty object of type UAParser.IResult because otherwise
+// UAParser returns the values for the current Browser. This is confusing because results are shown for an empty
+// UA field value.
+const getUserAgentInfo = (userAgent: string) =>
+  userAgent.trim().length > 0
+    ? UAParser(userAgent.trim())
+    : ({ ua: '', browser: {}, cpu: {}, device: {}, engine: {}, os: {} } as UAParser.IResult);
+const userAgentInfo = computed(() => withDefaultOnError(() => getUserAgentInfo(ua.value), undefined));
+
+const sections: {
+  heading: string;
+  icon?: PropType<any>;
+  content: {
+    label: string;
+    getValue: (blocks: UAParser.IResult) => string | undefined;
+    undefinedFallback?: string;
+  }[];
+}[] = [
```

Comment:
Can you mutualise and export this type in a file `user-agent-parser.types.ts` and reuse it in the result card component ? 
The idea is to prevent duplication of the same type definition and DRY the code